cmd/haqerBedComps/haqerBedComps.go

```diff
@@ -0,0 +1,194 @@
+package main
+
+//TODO : add check for self-overlap bed files
+//TODO : Remove file-name path in matrix output
+//TODO: add enrichment calculations
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/bed"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+	"strings"
+)
+
+type settings struct {
+	bedA             string
+	bedB             string
+	outFile          string
+	list             string
+	matrixAverage    string
+	matrixComponents string
+}
+
+type matrixLine struct {
+	name string
+	vals []float64
+}
+
+func compareTwo(s settings) {
+	var out []string
+	a := bed.Read(s.bedA)
+	b := bed.Read(s.bedB)
+	aNameSlice := strings.Split(s.bedA, "/")
+	aName := aNameSlice[len(aNameSlice)-1]
+	bNameSlice := strings.Split(s.bedB, "/")
+	bName := bNameSlice[len(bNameSlice)-1]
+	intervalsA := interval.BedSliceToIntervals(a)
+	intervalsB := interval.BedSliceToIntervals(b)
+	overlapsA, overlapsB, overlapAverage := interval.IntervalSimilarity(intervalsA, intervalsB)
+	header := fmt.Sprintf("percent overlaps of %s in %s\tpercent overlaps of %s in %s\tbedSimilarityScore", aName, bName, bName, aName)
+	data := fmt.Sprintf("%f\t%f\t%f", overlapsA, overlapsB, overlapAverage)
+	out = append(out, header)
+	out = append(out, data)
+	fileio.Write(s.outFile, out)
+}
+
+func multipleComparisons(s settings) {
+	var a, b []bed.Bed
+	var intervalsA, intervalsB []interval.Interval
+	var percA, percB, avg float64
+	var outMatrix *fileio.EasyWriter
+	var err error
+	var j int
+	var allFiles []string = []string{"x"}
+
+	out := fileio.EasyCreate(s.outFile)
+	fileio.WriteToFileHandle(out, "A\tB\tpercent overlaps of A in B\tpercent overlaps of B in A\tbedSimilarityScore")
+
+	if s.matrixAverage != "" {
+		outMatrix = fileio.EasyCreate(s.matrixAverage)
+	}
+	if s.matrixComponents != "" {
+		outMatrix = fileio.EasyCreate(s.matrixComponents)
+	}
+
+	beds := fileio.Read(s.list)
+	for i := range beds {
+		var l matrixLine
+		l.name = beds[i]
+		a = bed.Read(beds[i])
+		intervalsA = interval.BedSliceToIntervals(a)
+		for j = range beds {
+			if beds[i] == beds[j] {
+				//fileio.WriteToFileHandle(out, fmt.Sprintf("%s\t%s\t%f\t%f\t%f", beds[i], beds[j], 1.0, 1.0, 1.0))
+				if s.matrixComponents != "" || s.matrixAverage != "" {
+					l.vals = append(l.vals, 1.0)
```

Comment:
The matrix components are only if you want to make a heat-map. In the default output which is just a table of values, i decided it would not be necessary to output the self comparisons, which are all just 1